include/deal.II/base/aligned_vector.h

```diff
@@ -468,6 +556,97 @@ namespace internal
     }
   };
 
+
+
+  /**
+   * Class that issues the set commands for AlignedVector.
+   *
+   * @tparam initialize_memory Sets whether the set command should
+   * initialize memory (with a call to the copy constructor) or rather use the
+   * copy assignment operator. A template is necessary to select the
+   * appropriate operation since some classes might define only one of those
+   * two operations.
+   *
+   * @relates AlignedVector
+   */
+  template <typename T, bool initialize_memory>
+  class AlignedVectorDefaultInitialize : private parallel::ParallelForInteger
+  {
+    static const std::size_t minimum_parallel_grain_size = 160000/sizeof(T)+1;
+  public:
+    /**
+     * Constructor. Issues a parallel call if there are sufficiently many
+     * elements, otherwise work in serial.
+     */
+    AlignedVectorDefaultInitialize (const std::size_t size,
+                                    T *destination)
+      :
+      destination_ (destination),
+      trivial_element (false)
+    {
+      if (size == 0)
+        return;
+
+      // do not use memcmp for long double because on some systems it does not
+      // completely fill its memory and may lead to false positives in
+      // e.g. valgrind
+      if (std::is_trivial<T>::value == true &&
+          std::is_same<T,long double>::value == false)
+        {
+          const unsigned char zero [sizeof(T)] = {};
+          // cast element to (void*) to silence compiler warning for virtual
+          // classes (they will never arrive here because they are
+          // non-trivial).
+          T element {};
+          if (std::memcmp(zero, (void *)&element, sizeof(T)) == 0)
+            trivial_element = true;
+        }
```

Comment:
@bangerth We have two failing tests and after some investigation I found out the cause: We get undefined memory for `long double` because `new (&destination_[i]) T;` in line 646 does not do anything for a trivial type. We should skip the whole block in lines 590 to 603 here, delete the variable `trivial_element` from the class and only check for `std::is_trivial<T>::value` in line 620: After all, we know that for a trivial type the neutral element is simply zeros in memory, so we can use `memset` and do not need `memcmp` here in the default initialize class that we used in the generic set function.

Reply:
Just to avoid you to look it up: the failing tests were `full_matrix/full_matrix_10.debug` and `full_matrix/full_matrix_11.debug`, and they failed wiith floating point exception/uninitialized memory in the `long double` branch.